### PR TITLE
[SPARK-46377][INFRA] Upgrade labeler action to v5

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to upgrade `labeler action` from v4 to v5.

### Why are the changes needed?
The full release notes: https://github.com/actions/labeler/releases/tag/v5.0.0
Version 5 of this action updated the [runtime to Node.js 20](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runs-for-javascript-actions). All scripts are now run with Node.js 20 instead of Node.js 16 and are affected by any breaking changes between Node.js 16 and 20.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Pass GA.
- Manually check:
   1.Considering that the testing of the `github labeler action` is relatively special: When I submitted changes to this repo, the GA still executed the v4 version logic before merging the PR into it
   ref: https://github.com/actions/labeler?tab=readme-ov-file#notes-regarding-pull_request_target, 
   image

  2.I created a similar `private repo` to verify it
  The file (`.github/labeler.yml`) of this private repo is exactly the same as the file I am submitting to Spark repo
  image
  https://github.com/panbingkun/github-action-test/blob/main/.github/labeler.yml

  3.The verification process is as follows:
  A.When I submitted `core/x.scala` to this repo, it automatically labeled `CORE`
   https://github.com/panbingkun/github-action-test/pull/2
   image

   B.When I submitted `core/xxUI. scala` in this library, it automatically labeled `WEBUI`
   https://github.com/panbingkun/github-action-test/pull/3
   image

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @yaooqinn @LuciferYang

File: .github/labeler.yml [hunk 1]
```diff
@@ -18,109 +18,217 @@
...
-  - ".github/**/*"
-  - "appveyor.yml"
-  - "tools/**/*"
-  - "dev/create-release/**/*"
-  - ".asf.yaml"
-  - ".gitattributes"
-  - ".gitignore"
-  - "dev/merge_spark_pr.py"
-  - "dev/run-tests-jenkins*"
+  - changed-files:
```
Comment (panbingkun) [inline .github/labeler.yml:21]: ref: https://github.com/actions/labeler?tab=readme-ov-file#usage

File: .github/labeler.yml [hunk 2]
```diff
@@ -18,109 +18,217 @@
...
+
 CORE:
-  - any: ["core/**/*", "!**/*UI.scala", "!**/ui/**/*"]
-  - "common/kvstore/**/*"
-  - "common/network-common/**/*"
-  - "common/network-shuffle/**/*"
-  - "python/pyspark/*.py"
-  - "python/pyspark/tests/**/*.py"
+  - changed-files:
+    - all-globs-to-any-file: [
```
Comment (panbingkun) [inline .github/labeler.yml:69]: `any` -> `all-globs-to-any-file`
The logic of `all-globs-to-any-file` has been validated in private repo using `https://github.com/panbingkun/github-action-test/pull/2` and `https://github.com/panbingkun/github-action-test/pull/3`

**Comment by panbingkun** [comment]
ref: https://github.com/actions/labeler?tab=readme-ov-file#usage

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

Thank you for your effort on this non-trivial changes, @panbingkun .

Merged to master for Apache Spark 4.

**Pull request closed without merging**